src/test/java/com/nihongo/EmployeeControllerTest.java

```diff
@@ -0,0 +1,187 @@
+
+        EmployeePostRequest employeePostRequest = new EmployeePostRequest();
+        employeePostRequest.setUsername("test-employee");
+        employeePostRequest.setEmail("test-email");
+        employeePostRequest.setDob("0 month");
+        employeePostRequest.setAddress("test-add");
+
+        Employee employee = new Employee();
+        employee.setEmployeeId(Long.parseLong("1"));
+        employee.setUsername("test-employee");
+        employee.setMail("test-email");
+        employee.setDob("0 month");
+        employee.setAddress("test-add");
+
+        when(employeeService.saveEmployee(any())).thenReturn(employee);
+
+        MockHttpServletResponse response = mockMvc.perform(post("/employees")
+                .contentType("application/json")
+                .content(objectMapper.writeValueAsString(employeePostRequest))
+                .accept("application/json")).andReturn().getResponse();
+
+        Assertions.assertEquals(200, response.getStatus());
+
+        Map map = objectMapper.readValue(response.getContentAsString(), Map.class);
+        Assertions.assertEquals(1,map.get("id"));
+//        Employee employee1 = objectMapper.readValue(response.getContentAsString(), Employee.class);
+//        Assertions.assertEquals(1, employee1.getEmployeeId());
+    }
+
+    @Test
+    void putTest() throws Exception {
+        EmployeePutRequest employeePutRequest = new EmployeePutRequest();
+        employeePutRequest.setUsername("Sidd");
+        employeePutRequest.setMail("mohd.shoaib@gmail.com");
+        employeePutRequest.setDob("20-06-1990");
+        employeePutRequest.setAddress("Bareilly");
+
+        Employee employee = new Employee();
+        employee.setEmployeeId(Long.parseLong("1"));
+        employee.setUsername("Sidd");
+        employee.setMail("mohd.shoaib@gmail.com");
+        employee.setDob("20-06-1990");
+        employee.setAddress("Bareilly");
+
+        when(employeeService.updateEmployee(any(), any())).thenReturn(employee);
+
+        MockHttpServletResponse response = mockMvc.perform(put("/employees?id=1")
+                .contentType("application/json")
+                .content(objectMapper.writeValueAsString(employeePutRequest))
+                .accept("application/json")).andReturn().getResponse();
+
+
+        Assertions.assertEquals(200, response.getStatus());
+        Map map = objectMapper.readValue(response.getContentAsString(), Map.class);
+        Assertions.assertEquals(1,map.get("id"));
+    }
+
+    @Test
+    void deleteEmployeeTest() throws Exception {
+        EmployeeResponse employeeResponse = new EmployeeResponse();
+        employeeResponse.setId(Long.parseLong("1"));
+        employeeResponse.setUsername("Sidd");
+        employeeResponse.setMail("mohd.shoaib@gmail.com");
+        employeeResponse.setDob("20-06-1990");
+        employeeResponse.setAddress("Bareilly");
+
+        Employee employee = new Employee();
+        employee.setEmployeeId(Long.parseLong("1"));
```

Comment:
Are you using this object anywhere in this test case? if not remove it

Reply:
using this object